index_impl.go

```diff
@@ -228,7 +229,7 @@ func openIndexUsing(path string, runtimeConfig map[string]interface{}) (rv *inde
 		return nil, err
 	}
 
-	var im IndexMapping
+	var im *mapping.IndexMappingImpl
```

Comment:
I think it has to be exposed anyway, because it has lots of public members that users set directly.  So, in order to not break that, I don't return the interface, I return the actual implementation.

I'm not sure how you mean about moving the mapping unmarshaling...